gazebo/sensors/DepthCameraSensor.cc

```diff
@@ -124,6 +124,20 @@ void DepthCameraSensor::Init()
     gzerr << "No world name" << std::endl;
   }
 
+  if (this->dataPtr->depthCamera->GetOutputReflectance())
+  {
+    ignition::transport::AdvertiseMessageOptions opts;
+    opts.SetMsgsPerSec(50);
```

Comment:
should we use the `Sensor::UpdateRate()` instead of hard-coding to 50 Hz?

Reply:
I noticed that the other cameras are using a hard-coded 50Hz:

https://github.com/osrf/gazebo/blob/26b69a7cff29312164558d23344eb551a96508ba/gazebo/sensors/CameraSensor.cc#L110-L112

https://github.com/osrf/gazebo/blob/26b69a7cff29312164558d23344eb551a96508ba/gazebo/sensors/WideAngleCameraSensor.cc#L146-L148

When I tried changing the `50` to `this->UpdateRate()`, I get a broken topic because I think the update rate has not been set and still is the value 0.000. 

I could change this and initialize the update rate before advertising the topic but I am unsure if this will cause problems to the other components on the depth camera.

Reply:
`gazebo::transport::Node::Advertise` has [two optional numeric parameters](https://github.com/osrf/gazebo/blob/gazebo11_11.4.0/gazebo/transport/Node.hh#L176-L179), first the queue size and then the rate of publishing in Hz. In these examples from CameraSensor and WideAngleCameraSensor, `50` is actually the queue size. In the ignition::transport code (that has now been removed) it was a bit more clear that `SetMsgsPerSec` was about the rate of publishing

I think no additional changes are needed to the queue size